apps/hosts/views.py

```diff
@@ -162,7 +162,7 @@ def challenge_host_detail(request, challenge_host_team_pk, pk):
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
     try:
-        challenge_host = ChallengeHost.objects.get(pk=pk)
+        challenge_host = ChallengeHost.objects.get(pk=challenge_host_pk)
```

Comment:
Can we please use the `get_model_object` function here to fetch the object. For reference, please see - https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/utils.py#L25

Reply:
@RishabhJain2018 Sir, when i used get_model_object, one of the test failed as the error now is thrown is in different format i.e it's attaching id also along with other details like code_error. Any suggestions regarding, how can i deal with it sir as i am unable to fix it.

Reply:
@Kajol-Kumari You can edit the tests accordingly. 
For example, see [here](https://github.com/Cloud-CV/EvalAI/blob/master/tests/unit/challenges/test_views.py#L359-L369).
As an improvement on the mentioned code, you can store the invalid pk as `self.invalid_challenge_pk` and use it in `kwargs` and `expected`.
cc: @RishabhJain2018

Reply:
@nikochiko the problem is i am unable to get that ErrorDetail string as when i am trying to access it, it is showing undefined "detail", undefined "ErrorDetail"

Reply:
Please see the new comment below.